etna/transforms/nn/pytorch_forecasting.py

```diff
@@ -123,9 +115,12 @@ def fit(self, df: pd.DataFrame) -> "PytorchForecastingTransform":
             for feature_name in self.time_varying_known_categoricals:
                 df_flat[feature_name] = df_flat[feature_name].astype(str)
 
-        freq_unit = self._calculate_freq_unit(self.freq)
-        df_flat["time_idx"] = (df_flat["timestamp"] - self.min_timestamp) / freq_unit
-        df_flat["time_idx"] = df_flat["time_idx"].astype(int)
+        # making time_idx feature.
+        # it's needed for pytorch-forecasting for proper train-test split.
+        # it should be incremented by 1 for every new timestamp.
+        df_flat["time_idx"] = (df_flat["timestamp"] - self.min_timestamp) // pd.Timedelta("1s")
```

Comment:
Why do we need to keep only remainder of the division?

Reply:
It's casting timestamp delta to integer.